cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -246,7 +248,226 @@ EXEC strQry // SQL Injection vulnerability!
+|-----------|:-----:|
+|Comma	|,|
+|Backslash character|\|
+|Pound sign (hash sign)|#|
+|Plus sign|+|
+|Less than symbol|<|
+|Greater than symbol|>|
+|Semicolon|;|
+|Double quote (quotation mark)|"|
+|Equal sign|=|
+|Leading or trailing spaces| | 
+
+NB: The space character must be escaped only if it is the leading or trailing character in a component name, such as a Common Name. Embedded spaces should not be escaped.
+
+## A2 Broken Authentication
+
+DO: Use [ASP.net Core Identity](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity?view=aspnetcore-2.2&).
+ASP.net Core Identity framework is well configured by default, where it uses secure password hashes and an individual salt. Identity uses the PBKDF2 hashing function for passwords, and they generate a random salt per user.
+
+DO: Set secure password policy
+
+e.g ASP.net Core Identity 
+
+``` csharp
+//startup.cs
+services.Configure<IdentityOptions>(options =>
+{
+	// Password settings
+	options.Password.RequireDigit = true;
+	options.Password.RequiredLength = 8;
+	options.Password.RequireNonAlphanumeric = true;
+	options.Password.RequireUppercase = true;
+	options.Password.RequireLowercase = true;
+	options.Password.RequiredUniqueChars = 6;
+ 
+ 
+	options.Lockout.DefaultLockoutTimeSpan = TimeSpan.FromMinutes(30);
+	options.Lockout.MaxFailedAccessAttempts = 3;
+ 
+	options.SignIn.RequireConfirmedEmail = true;
+	
+	options.User.RequireUniqueEmail = true;
+});
+```
+
+DO: Set a cookie policy
+
+e.g 
+
+``` csharp
+//startup.cs
+services.ConfigureApplicationCookie(options =>
+{
+	options.Cookie.HttpOnly = true;
+	options.Cookie.Expiration = TimeSpan.FromHours(1)
+	options.SlidingExpiration = true;
+});
+```
+
+## A3 Sensitive Data Exposure
+
+DO NOT: [Store encrypted passwords](Password_Storage_Cheat_Sheet.md#do-not-limit-the-character-set-and-set-long-max-lengths-for-credentials).
+
+DO: Use a strong hash to store password credentials. Use Argon2, PBKDF2, BCrypt or SCrypt with at least 8000 iterations and a strong key.
+
+DO: Enforce passwords with a minimum complexity that will survive a dictionary attack i.e. longer passwords that use the full character set (numbers, symbols and letters) to increase the entropy.
+
+DO: Use a strong encryption routine such as AES-512 where personally identifiable data needs to be restored to it's original format. Do not encrypt passwords. Protect encryption keys more than any other asset. Apply the following test: Would you be happy leaving the data on a spreadsheet on a bus for everyone to read. Assume the attacker can get direct access to your database and protect it accordingly. More information [here](Transport_Layer_Protection_Cheat_Sheet.md).
+
+DO: Use TLS 1.2 for your entire site. Get a free certificate [LetsEncrypt.org](https://letsencrypt.org/).
+
+DO NOT: [Allow SSL, this is now obsolete](https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices#22-use-secure-protocols).
```

Comment:
Point to Transport Layer Protection CS to keep advices up to dates.

Reply:
Still missing when I take the content of your [fork](https://github.com/knadt/CheatSheetSeries/blob/master/cheatsheets/DotNet_Security_Cheat_Sheet.md)